posthog/api/dashboards/test/test_dashboard_templates.py

```diff
@@ -77,6 +122,126 @@ def setUp(self):
+
+        assert DashboardTemplate.objects.count() == 1
+        id = DashboardTemplate.objects.first().id
+
+        response = self.client.get(f"/api/projects/{self.team.pk}/dashboard_templates/{id}")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response)
+
+        keys_to_check = ["template_name", "dashboard_description", "tags", "variables", "tiles", "dashboard_filters"]
+
+        for key in keys_to_check:
+            assert response.json()[key] == variable_template[key], f"key {key} failed"
+
+    def test_delete_dashboard_template_by_id(self) -> None:
+        response = self.client.post(
+            f"/api/projects/{self.team.pk}/dashboard_templates",
+            variable_template,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response)
+
+        assert DashboardTemplate.objects.count() == 1
+        id = DashboardTemplate.objects.first().id
+
+        response = self.client.patch(f"/api/projects/{self.team.pk}/dashboard_templates/{id}", {"deleted": True})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response)
+
+        response = self.client.get(f"/api/projects/{self.team.pk}/dashboard_templates")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response)
+
+        assert response.json()["results"] == [], response.json()
+
+    def test_non_staff_user_cannot_delete_dashboard_template_by_id(self) -> None:
+
+        response = self.client.post(
+            f"/api/projects/{self.team.pk}/dashboard_templates",
+            variable_template,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response)
+
+        assert DashboardTemplate.objects.count() == 1
+        id = DashboardTemplate.objects.first().id
+
+        self.user.is_staff = False
+        self.user.save()
+
+        response = self.client.patch(f"/api/projects/{self.team.pk}/dashboard_templates/{id}", {"deleted": True})
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN, response)
+
+        response = self.client.get(f"/api/projects/{self.team.pk}/dashboard_templates")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response)
+
+        assert response.json()["results"] != [], response.json()
+
+    def test_update_dashboard_template_by_id(self) -> None:
+        response = self.client.post(
+            f"/api/projects/{self.team.pk}/dashboard_templates",
+            variable_template,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        assert DashboardTemplate.objects.count() == 1
+        id = DashboardTemplate.objects.first().id
+
+        response = self.client.patch(
+            f"/api/projects/{self.team.pk}/dashboard_templates/{id}",
+            {"template_name": "new name"},
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response)
```

Comment:
We should prefer `assert x == y` format to copy pytest standard

(or at least try and use format per test file)

Only a nit since I think we area all chopping and changing